jvm/ca.ualberta.maple.swan.drivers/src/resources/models.swirl

```diff
@@ -199,19 +211,505 @@ bb0(%0 : $`*Array<τ_0_0>`, %1 : $`Array<τ_0_0>`, %2 : $`@thin Array<τ_0_0>.Ty
+bb0(%0 : $`*ArraySlice<τ_0_0>`, %1 : $`τ_0_0`, %2 : $`*Array<τ_0_0>`) :
+  arr = pointer_read %2, $`Array<τ_0_0>`
+  elem = pointer_read arr, $`τ_0_0`
+  pointer_write arr to %0
+  return elem
+}
+
+// %0 -> slice of array , %1 -> Maximum array length, %2 -> array
+//remove certain elements started from the end of the array
+func [model] @`(extension in Swift):Swift.BidirectionalCollection.dropLast(Swift.Int) -> A.SubSequence` : $`@out τ_0_0.SubSequence` {
+bb0(%0 : $`*ArraySlice<τ_0_0>`, %1 : $`τ_0_0`, %2 : $`*Array<τ_0_0>`) :
+  arr = pointer_read %2, $`Array<τ_0_0>`
+  elem = pointer_read arr, $`τ_0_0`
+  pointer_write arr to %0
+  return elem
+}
+
+// %0 -> sorted array
+// sort the array in certain order
+func [model] @`(extension in Swift):Swift.Sequence< where A.Element: Swift.Comparable>.sorted() -> [A.Element]` : $`@owned Array<τ_0_0.Element>` {
+bb0(%0 : $`*Array<τ_0_0>`):
+  arr = pointer_read %0, $`Array<τ_0_0>`
+  return arr
+}
+
+// %0 -> first String, %1 -> second String, %2 -> String type
+// compared the elements based on some conditions
+func [model] @`static (extension in Swift):Swift.Comparable.<= infix(A, A) -> Swift.Bool` : $`Bool` {
+bb0(%0 : $`*String`, %1 : $`*String`, %2 : $`@thick String.Type`):
+  %ret = new $`Bool`
+  return %ret
+}
+
+// %0 -> first String, %1 -> second String, %2 -> String type
+// compared the elements based on some conditions
+func [model] @`static Swift.String.< infix(Swift.String, Swift.String) -> Swift.Bool` : $`Bool` {
+bb0(%0 : $`String`, %1 : $`String`, %2 : $`String.Type`):
+  %ret = new $`Bool`
+  return %ret
+}
+
+// %0 -> first String, %1 -> second String, %2 -> String type
+// compared the elements based on some conditions
+func [model] @`static (extension in Swift):Swift.Comparable.>= infix(A, A) -> Swift.Bool` : $`Bool` {
+bb0(%0 : $`*String`, %1 : $`*String`, %2 : $`String.Type`):
+  %ret = new $`Bool`
+  return %ret
+}
+
+// %0 -> bool from condition to reorder the array , %1 -> array
+// read array from %1
+func [model] @`(extension in Swift):Swift.Sequence.sorted(by: (A.Element, A.Element) throws -> Swift.Bool) throws -> [A.Element]` : $`(@owned Array<τ_0_0.Element>, @error Error)` {
+bb0(%0 : $`(Bool, @error Error)`, %1 : $`*Array<τ_0_0>`):
+  arr = new $`(@owned Array<τ_0_0.Element>, @error Error)`
+  arr1 = pointer_read %1, $`Array<τ_0_0>`
+  val = pointer_read arr1, $`τ_0_0`
+  pointer_write val to arr
+  return arr
+}
+
+// %0 -> array
+// read array from %0
+func [model] @`(extension in Swift):Swift.Sequence.shuffled() -> [A.Element]` : $`@owned Array<τ_0_0.Element>` {
+bb0(%0 : $`*Array<τ_0_0>`):
+  arr = pointer_read %0, $`Array<τ_0_0>`
+  return arr
+}
+
+func [model] @`(extension in Swift):Swift.MutableCollection< where A: Swift.BidirectionalCollection>.reverse() -> ()` : $`()` {
+bb0(%0 : $`*Array<τ_0_0>`):
+  %ret = new $`()`
+  return %ret
+}
+
+// %0 -> optional String, %1 -> ptr to Array
```

Comment:
Be careful with the description of variables. `%0` actually represents the function given to `compactMap`.